### PR TITLE
Add recipes

### PR DESCRIPTION
**Pull request by RJMW**
* bioconductor-pmp
* bioconductor-struct
* bioconductor-structtoolbox
* r-xlsxbuilder

### CODE REVIEW

File: recipes/bioconductor-pmp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+build:
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+  noarch: generic
+# Suggests: testthat, covr, knitr, rmarkdown, BiocStyle, gridExtra, magick
+requirements:
+  host:
+    - 'bioconductor-impute >=1.64.0,<1.65.0'
```
Comment (grlloyd) [inline recipes/bioconductor-pmp/meta.yaml:21]: I dont think we need to be so specific here? bioc 3.13 will ensure the correct version is installed? Or do we need to specify the version that corresponds to bioc 3.13?

File: recipes/bioconductor-struct/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  rpaths:
+    - lib/R/lib/
+    - lib/
+  noarch: generic
+# Suggests: testthat, rstudioapi, rmarkdown, covr, BiocStyle, openxlsx, ggplot2, magick
+requirements:
+  host:
+    - 'bioconductor-s4vectors >=0.28.0,<0.29.0'
+    - 'bioconductor-summarizedexperiment >=1.20.0,<1.21.0'
+    - r-base >4.0
```
Comment (grlloyd) [inline recipes/bioconductor-struct/meta.yaml]: Needs to be R 4.1 for Bioc 3.13 for all packages

File: recipes/bioconductor-structtoolbox/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+build:
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+requirements:
+  host:
+    - pandoc
+    - icu
+    - r-base >4.0
```
Comment (grlloyd) [inline recipes/bioconductor-structtoolbox/meta.yaml]: R 4.1

File: recipes/bioconductor-structtoolbox/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    - r-rtsne
+    - r-stringi
+    - bioconductor-impute
+    - bioconductor-pcaMethods
+    - bioconductor-struct
+    - bioconductor-pmp
+  run:
+    - pandoc
+    - icu
+    - r-base >4.0
```
Comment (grlloyd) [inline recipes/bioconductor-structtoolbox/meta.yaml]: R 4.1

File: recipes/bioconductor-struct/meta.yaml (hunk 2 above)
Comment (RJMW) [reply recipes/bioconductor-struct/meta.yaml]: After discussing this we decided to stick with the specified version. We will use the conda build within bioc 3.12 but use the source from 3.13.  r-base needs pinning as follows: 'r-base >4.0,<4.1'

File: recipes/bioconductor-pmp/meta.yaml (hunk 1 above)
Comment (RJMW) [reply recipes/bioconductor-pmp/meta.yaml:21]: After discussing this we decided to stick with the specified version. We will use this conda build within bioc 3.12 but use the source from 3.13.

File: recipes/bioconductor-structtoolbox/meta.yaml (hunk 3 above)
Comment (RJMW) [reply recipes/bioconductor-structtoolbox/meta.yaml]: See above

File: recipes/bioconductor-structtoolbox/meta.yaml (hunk 4 above)
Comment (RJMW) [reply recipes/bioconductor-structtoolbox/meta.yaml]: See above

**Review by grlloyd — APPROVED** [review]

**Pull request merged**